tests/evm/test_extcodecopy.py

```diff
@@ -0,0 +1,176 @@
+import pytest
+from itertools import chain
+from zkevm_specs.copy_circuit import verify_copy_table
+from zkevm_specs.evm import (
+    verify_steps,
+    Tables,
+    Block,
+    Bytecode,
+    ExecutionState,
+    StepState,
+    Opcode,
+    U64,
+    U160,
+    RLC,
+    CopyCircuit,
+    CopyDataTypeTag,
+    AccountFieldTag,
+    CallContextFieldTag,
+)
+from zkevm_specs.evm.typing import RWDictionary
+from zkevm_specs.util import (
+    rand_bytes,
+    rand_fq,
+    keccak256,
+    EXTRA_GAS_COST_ACCOUNT_COLD_ACCESS,
+    GAS_COST_COPY,
+    memory_word_size,
+    memory_expansion,
+)
+
+TESTING_DATA = (
+    # empty code
+    (bytes(), True, True, 0x30000, 0x00, 0x00, 54),  # warm account
+    (bytes(), False, True, 0x30000, 0x00, 0x00, 54),  # cold account
+    # non-empty code
+    (bytes([10, 40]), True, True, 0x30000, 0x00, 0x00, 54),  # warm account
+    (bytes([10, 10]), False, True, 0x30000, 0x00, 0x00, 54),  # cold account
+    # code length > 256
+    (rand_bytes(64), True, True, 0x30000, 0x00, 0x00, 54),  # warm account
+    (rand_bytes(64), False, True, 0x30000, 0x00, 0x00, 54),  # cold account
+    # out of bound cases
+    (rand_bytes(64), True, True, 0x30000, 0x20, 0x00, 260),  # warm account
+    (rand_bytes(64), False, True, 0x30000, 0x20, 0x00, 260),  # cold account
+)
+
+
+@pytest.mark.parametrize(
+    "code, is_warm, is_persistent, address, src_addr, dst_addr, length", TESTING_DATA
+)
+def test_extcodecopy(
+    code: bytes,
+    is_warm: bool,
+    is_persistent: bool,
+    address: U160,
+    src_addr: U64,
+    dst_addr: U64,
+    length: U64,
+):
+    randomness = rand_fq()
+
+    code_hash = int.from_bytes(keccak256(code), "big")
+
+    next_memory_word_size = memory_word_size(dst_addr + length)
+    _, memory_expansion_cost = memory_expansion(0, dst_addr + length if length else 0)
+    memory_gas_cost = memory_expansion_cost + memory_word_size(length) * GAS_COST_COPY
+
+    gas_cost_extcodecopy = (
+        Opcode.EXTCODECOPY.constant_gas_cost()
+        + memory_gas_cost
+        + (not is_warm) * EXTRA_GAS_COST_ACCOUNT_COLD_ACCESS
+    )
+
+    tx_id = rand_fq()
+    call_id = rand_fq()
```

Comment:
As @ed255 and @CPerezz pointed out, may be you can use:
```suggestion
    tx_id = 2
    call_id = 3
```
so that test case failures will be reproducible.